src/datastoreapi/DataStoreAPI.java

```diff
@@ -48,20 +56,32 @@ public List<String> getOutputList() {
 
 	// setters
 	public void setInputList(List<Integer> inputList) {
+		if(inputList == null || inputList.isEmpty()) {
+			throw new IllegalArgumentException("Input list cannot be null or empty");
+		}
 		this.inputList = inputList;
 	}
 
 	public void setOutputList(List<String> outputList) {
+		if(outputList == null) {
+			throw new IllegalArgumentException("Output list cannot be null");
+		}
 		this.outputList = outputList;
 	}
 
 	// addInput method
 	public List<Integer> addInput(int a) {

```

Comment:
This method does nothing. Consider deleting it or changing it to do something. Maybe have it add to your inputList and use it in readInput?

Reply:
i'll keep it for now since it's just for the instance variable and i must of used it somewhere since the checks fail when i delete it. I'll make it actually do something in the future.